test/utils/googletest/CMakeLists.txt

```diff
@@ -16,7 +16,7 @@
 
 include(CheckCXXCompilerFlag)
 
-include_directories(SYSTEM 
+set(GTCLANG_GTEST_INCLUDE_DIRS 

```

Comment:
what is the difference between the include_directories(SYSTEM X) and target_include_directories(target X)?

Reply:
so include_directories is a bad practice. It will add `-I/dir` to every target compilation. 
While `target_include_directories` will only add it for this target. 
Since we need these headers not only for compiling gtest, but for all the unittest you have in the application using gtest, we declare this `PUBLIC` and automatically the headers willl be included in all targets that are linking against gtest target